gomock/argument_captor_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gomock_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/golang/mock/gomock/internal/mock_gomock"
+)
+
+const (
+	intArg = 45982
+	item1  = "item1"
+	item2  = "item2"
+)
+
+var sliceArg = []string{item1, item2}
+
+func TestValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockMatcher := mock_gomock.NewMockMatcher(ctrl)
+	captor := gomock.Captor(mockMatcher)
+
+	mockMatcher.EXPECT().Matches(intArg).Times(1)
+	mockMatcher.EXPECT().Matches(sliceArg).Times(1)
+
+	captor.Matches(intArg)
+	captor.Matches(sliceArg)
+
+	actualValue := captor.Value().([]string)
```

Comment:
Given the test, I know this will work, but lets check the type and error if it does not match. Will be useful if this test ever got refactored.